src/transformation/visitors/helpers/multi.ts

```diff
@@ -0,0 +1,170 @@
+import * as ts from "typescript";
+import * as lua from "../../../LuaAST";
+import * as utils from "../../utils/helpers";
+import { TransformationContext } from "../../context";
+import { transformAssignmentLeftHandSideExpression } from "../binary-expression/assignments";
+import { transformIdentifier } from "../identifier";
+import { transformArguments } from "../call";
+import { InvalidMultiHelperFunctionUse, UnsupportedKind } from "../../utils/errors";
+import { getDependenciesOfSymbol, createExportedIdentifier } from "../../utils/export";
+import { createLocalOrExportedOrGlobalDeclaration } from "../../utils/lua-ast";
+
+function isMultiHelperDeclaration(declaration: ts.Declaration): boolean {
+    return utils.getHelperFileKind(declaration.getSourceFile()) === utils.HelperKind.Multi;
+}
+
+function isMultiHelperCallSignature(context: TransformationContext, expression: ts.CallExpression): boolean {
+    const type = context.checker.getTypeAtLocation(expression.expression);
+    return type.symbol?.declarations?.some(isMultiHelperDeclaration) ?? false;
+}
+
+function isMultiReturningCallExpression(context: TransformationContext, expression: ts.CallExpression): boolean {
+    const signature = context.checker.getResolvedSignature(expression);
+    return signature?.getReturnType().aliasSymbol?.declarations?.some(isMultiHelperDeclaration) ?? false;
+}
+
+export function isMultiHelperNode(context: TransformationContext, node: ts.Node): boolean {
+    const type = context.checker.getTypeAtLocation(node);
+    return type.symbol?.declarations?.some(isMultiHelperDeclaration) ?? false;
+}
+
+export function transformMultiHelperReturnStatement(
+    context: TransformationContext,
+    statement: ts.ReturnStatement
+): lua.Statement | undefined {
+    if (!statement.expression) {
+        return undefined;
+    }
+
+    if (!ts.isCallExpression(statement.expression)) {
+        return undefined;
+    }
+
+    if (!isMultiHelperCallSignature(context, statement.expression)) {
+        return undefined;
+    }
+
+    const expressions = transformArguments(context, statement.expression.arguments);
+    return lua.createReturnStatement(expressions, statement);
```

Comment:
How would this work with spread (especially after #832)?

Reply:
Current plan:

You can spread the result of a function that calls $multi.

```ts
[...callMulti(1, 2), 3]; // [1, 2, 3]
```

But you can't spread a $multi call.

```ts
[...$multi(1, 2), 3]; // error, use [1, 2, 3] instead
```